### PR TITLE
Review Register Allocation Tests

### PR DESCRIPTION
**Pull request by vext01**
See individual commits.

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -47,6 +47,10 @@ lazy_static! {
                                      R10.code(), RBX.code(), R12.code(), R13.code(), R14.code(),
                                      R15.code()];
 
+    // The trace inputs/output are always allocated to this reserved register.
```
Comment (vext01) [inline ykcompile/src/lib.rs:50]: output*s*

**Comment by ltratt** [comment]
Overall, to the extent I understand this (it's not code I'm very familiar with!) this looks good, and I'm pleased that tests haven't disappeared! My only comment is that when it comes to squashing/rebasing time, can we provide a bit more motivation in some of the commits (so not just "what" but also "why")? For example in https://github.com/softdevteam/yk/pull/184/commits/e005a61a50b5388fce0aa9c39d950a4a3cacbe0d inlining `init` probably just needs something like "Since we only call this once, there's no advantage to having it be a separate function"; in https://github.com/softdevteam/yk/pull/184/commits/177160b1ec7a8d154ce894395480463f0cff2aa7 I presume the implication is that "We didn't previously test this"?

**Comment by ptersilie** [comment]
Looks good to me, though agree with what Laurie said.

**Comment by vext01** [comment]
Sure, I'll expand the commit messages.

OK to fix that typo and squash at the same time?

**Comment by ltratt** [comment]
Please go ahead.

**Comment by ptersilie** [comment]
Sure.

**Comment by vext01** [comment]
Give this a go.

**Comment by ptersilie** [comment]
Looks good to me.

**Pull request merged**